multiview/embed/splitae.py

```diff
@@ -0,0 +1,209 @@
+import torch
+from torch.utils.data import Dataset, DataLoader
+from torchvision import datasets
+import torchvision
+import matplotlib.pyplot as plt
+import PIL
+import numpy as np
+import itertools
+import tqdm
+
+from .base import BaseEmbed
+from ..utils.utils import check_Xs
+
+
+class FullyConnectedNet(torch.nn.Module):
+    def __init__(self, inputSize, hiddenSize, numHiddenLayers, embeddingSize):
+        super().__init__()
+        assert numHiddenLayers >= 0, "can't have negative hidden layer count"
+        self.layers = torch.nn.ModuleList()
+        if numHiddenLayers == 0:
+            self.layers.append(torch.nn.Linear(inputSize, embeddingSize))
+        else:
+            self.layers.append(torch.nn.Linear(inputSize, hiddenSize))
+            for i in range(numHiddenLayers-1):
+                self.layers.append(torch.nn.Linear(hiddenSize, hiddenSize))
+            self.layers.append(torch.nn.Linear(hiddenSize, embeddingSize))
+
+    def forward(self, x):
+        for layer in self.layers[:-1]:
+            x = torch.nn.Sigmoid()(layer(x))
+        x = self.layers[-1](x)  # no activation on last layer
+        return x
+
+    def paramCount(self):
+        return np.sum([np.prod(s.shape) for s in self.parameters()])
+
+
+class SplitAE(BaseEmbed):
+    """
+    Implements an autoencoder that creates an embedding of a view View1 and
+    from that embedding reconstructs View1 and another view View2.
+    Parameters
```

Comment:
need spacing between these things to get to render properly, look at some of the graspy docs for examples